lib/js/components/ProgressBar/ProgressBar.vue

```diff
@@ -0,0 +1,330 @@
+<template>
+	<div
+		:class="{
+			progressBar: true,
+			'-primary': color === PROGRESS_BAR_COLORS.PRIMARY,
+			'-warning': color === PROGRESS_BAR_COLORS.WARNING,
+			'-fail': color === PROGRESS_BAR_COLORS.FAIL,
+			'-success': color === PROGRESS_BAR_COLORS.SUCCESS,
+			'-neutral': color === PROGRESS_BAR_COLORS.NEUTRAL,
+			'-info': color === PROGRESS_BAR_COLORS.INFO,
+			'-schemeMedium': colorScheme === PROGRESS_BAR_COLOR_SCHEMES.MEDIUM,
+			'-schemeMediumNeutral': colorScheme === PROGRESS_BAR_COLOR_SCHEMES.MEDIUM_NEUTRAL,
+			'-compact': layout === PROGRESS_BAR_LAYOUTS.COMPACT,
+		}"
+	>
+		<div class="progressBar__label">
+			<div class="progressBar__labelText">{{ labelText }}</div>
+			<div v-if="labelDataExist" class="progressBar__labelDataWrapper">
+				<div v-if="labelData" class="progressBar__labelData">{{ labelData }}</div>
+				<div v-if="labelDataSupporting" class="progressBar__labelDataSupporting">
+					<span class="progressBar__labelDataSeparator">/</span>
+					{{ labelDataSupporting }}
+				</div>
+				<div v-if="labelDataSuffix" class="progressBar__labelDataSuffix">{{
+					labelDataSuffix
+				}}</div>
+			</div>
+		</div>
+		<div
+			:class="{
+				progressBar__bar: true,
+				'-small': size === PROGRESS_BAR_SIZES.SMALL,
+				'-xsmall': size === PROGRESS_BAR_SIZES.XSMALL,
+				'-noRadius': radius === PROGRESS_BAR_RADII.NONE,
+			}"
+		>
+			<div class="progressBar__results">
+				<div
+					v-for="(range, index) in ranges"
+					:key="index"
+					:class="{
+						progressBar__result: true,
+						'-secondary': range.layer === 2 && numberOfLayers === 2,
+					}"
+					:style="{ left: range.percentValueFrom + '%', width: range.length + '%' }"
+				>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/typography';
+@import '../../../styles/settings/media-queries';
+@import '../../../styles/settings/colors/tokens';
+
+$progress-bar-height: 16px;
+$progress-bar-s-height: 8px;
+$progress-bar-xs-height: 4px;
+
+$progress-bar-border-radius: 8px;
+
+$progress-bar-layers: (
+	'primary': (
+		'default': $color-primary-text,
+		'ghost': $color-primary-data-ghost,
```

Comment:
instead of naming it based on color used, I'd name it based on usage, eg. `default-secondary-layer`. If you do that, there won't be any problems when someone defines non-ghoust color here. What do you think?

Reply:
It's good idea, but it's hard to hame these keys, what do you think about these keys:
```
		'default-layer': $color-primary-text,
		'default-layer-secondary-': $color-primary-data-ghost,
		'second-layer': $color-primary-data-medium,
		'second-layer-secondary': $color-primary-data-weak,
		'placeholder-layer': $color-neutral-data-weak,
```

Reply:
IMO, proposed key names are wrong:
`second-layer` uses color that is used for 1st data layer in `MEDIUM` color scheme

Please try again

Reply:
When I look at this screen:
![Screenshot 2022-09-30 at 11 45 37](https://user-images.githubusercontent.com/4208462/193243383-c3c03c93-4474-4257-b8db-fbd6550a7806.png)
and 3 layer of data, I dont see sense of changing these keys. If someone would use noghost color, he should add new key `noghost` and use it the new place or replace `ghost` by it.

Reply:
ok, let's try another game:
Imagine, we keep current colors config scheme. In 3 months UX wants us to change color for second layer in default color scheme from `ghost` to `strong`. What needs to be done?

Reply:
I understand it, but can't find correct word. Check below names, if they are wrong please help me:
```
'default-layer': $color-primary-data,
'default-layer-secondary': $color-primary-data-ghost,
'emphasize-layer': $color-primary-data-medium,
'emphasize-layer-secondary': $color-primary-data-weak,
'placeholder-layer': $color-neutral-data-weak,
```

Reply:
what would you say about
```
'default-color-scheme-first-layer`
'default-color-scheme-second-layer`
'medium-color-scheme-first-layer`
...
'medium-neutral-color-scheme-second-layer`
```
?

Reply:
Sounds good, but is `color-scheme` needed? It's only internal key for this variable in this single component. Without it is clear for me also. We haven't there any other colors so these are unambiguous for me.

Reply:
when I was yunior dev, I though I don't need to make variable names self explanatory - it's all visible in the code. Now I see, I was wrong. We read code much more often than we write it, so forcing someone to go through code once again to understand something is a waste of time. That's why I like longer names now.